src/main/scala/com/itechart/ny_accidents/service/WeatherMappingService.scala

```diff
@@ -41,6 +43,22 @@ class WeatherMappingService @Inject()(weatherDAO: WeatherDAO, weatherParser: Wea
     }
   }
 
+  def getWeatherByStationsBetweenDates(earlyDate: Long, laterDate: Long): Map[Int, Seq[WeatherEntity]] = {
+    val earlyDateHash = DateUtils.hashByDate(earlyDate)
+    val laterDateHash = DateUtils.hashByDate(laterDate)
+
+    (FIRST_STATION_ID to LAST_STATION_ID)
+      .map(stationId => {
+        val weatherMap = (earlyDateHash to laterDateHash by GeneralConstants.HASH_DIFFERENCE)
```

Comment:
It will be always the same as earlyDateHash and laterDateHash are not changed, you can move it out of map.

Reply:
Sorry, I don't understand this part

Reply:
I meant that `val weatherMap = (earlyDateHash to laterDateHash by GeneralConstants.HASH_DIFFERENCE)` is same for each stationId, so you don't to evaluate it each time

Reply:
Done